packages/ecdsa-secp256k1-verification-key-2019/src/verificationKey.ts

```diff
@@ -0,0 +1,235 @@
+import base64url from 'base64url'
+import createHash from 'create-hash'
+import * as secp256k1 from 'secp256k1'
+import randomBytes from 'randombytes'
+
+const cryptoLd = require('crypto-ld')
+const base58 = require('base58-universal')
+
+const SUITE_ID = 'EcdsaSecp256k1VerificationKey2019'
+
+const sha256 = (data: any) => createHash('sha256').update(data).digest()
+
+type ExportedKey = {
+  '@context'?: string
+  type: string
+  id: string
+  controller: string
+  publicKeyBase58?: string
+  privateKeyBase58?: string
+  revoked?: boolean
+}
+
+type EcdsaSecp256k1VerificationKey2019Options = {
+  controller: string
+  id: string
+  revoked?: boolean
+  publicKeyBase58?: string
+  privateKeyBase58?: string
+}
+
+type EcdsaSecp256k1VerificationKey2019HexKeyOptions = {
+  controller: string
+  id: string
+  revoked?: boolean
+  publicKeyHex?: string
+  privateKeyHex?: string
+}
+
+export class EcdsaSecp256k1VerificationKey2019 extends cryptoLd.LDKeyPair {
+  public type: string
+
+  public publicKeyBase58?: string
+
+  public privateKeyBase58?: string
+
+  constructor({ publicKeyBase58, privateKeyBase58, ...options }: EcdsaSecp256k1VerificationKey2019Options) {
+    super(options)
+
+    if (privateKeyBase58 && !publicKeyBase58) {
+      const publicKey = secp256k1.publicKeyCreate(base58.decode(privateKeyBase58))
+      this.publicKeyBase58 = base58.encode(publicKey)
+    } else {
+      this.publicKeyBase58 = publicKeyBase58
+    }
+
+    this.type = SUITE_ID
+    this.privateKeyBase58 = privateKeyBase58
+
+    if (!this.publicKeyBase58) {
+      throw new TypeError('The "publicKeyBase58" property is required.')
+    }
+  }
+
+  static from(options: EcdsaSecp256k1VerificationKey2019Options | EcdsaSecp256k1VerificationKey2019HexKeyOptions) {
+    if ((options as any).publicKeyHex || (options as any).privateKeyHex) {
+      const { publicKeyHex, privateKeyHex, ...rest } = options as any
+
+      return new EcdsaSecp256k1VerificationKey2019({
+        ...rest,
+        publicKeyBase58: publicKeyHex ? base58.encode(Buffer.from(publicKeyHex, 'hex')) : undefined,
+        privateKeyBase58: privateKeyHex ? base58.encode(Buffer.from(privateKeyHex, 'hex')) : undefined,
+      })
+    }
+
+    return new EcdsaSecp256k1VerificationKey2019(options)
+  }
+
+  static async generate({
```

Comment:
What's going on with this function? It seems that one could provide a `seed` value that could theoretically loop forever, no?

What condition causes `!secp256k1.privateKeyVerify(privateKey) || !secp256k1.publicKeyVerify(publicKey)` to be true?

Assuming the while loop is needed, I wonder if we could have a `let isVerifiable` used by the while loop instead of true / continue.

Reply:
Also, can we not `console.log` the private keys or only do it based on an environment variable?

Reply:
Oh yeah I'll remove the log. Yeah I'll fix it up so that if the provided seed is invalid it'll fail instead of infinite loop